disco_aws_automation/disco_vpc.py

```diff
@@ -750,68 +750,119 @@ def find_vpc_id_by_name(vpc_name):
-            """return type from `name[:type]/metanetwork`, defaulting to name if type is omitted"""
-            return endpoint.split('/')[0].split(':')[-1].strip()
-
-        def get_metanetwork(endpoint):
-            """return metanetwork from `name[:type]/metanetwork`"""
-            return endpoint.split('/')[1].strip()
+        Args:
+            line (str): A vpc peering config line of the form
+                            `vpc_name[:vpc_type]/metanetwork vpc_name[:vpc_type]/metanetwork`
 
-        def safe_get_from_list(_list, i):
-            """returns the i-th element in a list, or None if it doesn't exist"""
-            return _list[i] if _list and len(_list) > i else None
+                        `vpc_name` may be the name of a VPC or a `*` wildcard to peer with any VPC of vpc_type
+            existing_vpcs (List[VPC]): A list of Boto VPC objects for all the existing AWS VPCs
 
-        vpc_type_map = {
-            get_vpc_name(endpoint): get_vpc_type(endpoint)
-            for endpoint in endpoints
+        Returns:
+            data necessary to create peering connections for the VPCs as a dictionary record for each
+            peering connection containing two other dictionaries: vpc_name -> DiscoVPC instance
+            and vpc_name -> metanetwork name
+        """
+        logging.debug('checking existence for peering %s', line)
+        endpoints = line.split(' ')
+        if not len(endpoints) == 2:
+            raise VPCConfigError('Invalid peering config "%s". Peering config must be of the format '
+                                 'vpc_name[:vpc_type]/metanetwork vpc_name[:vpc_type]/metanetwork' % line)
+
+        def get_endpoint_info(endpoint):
+            """ Get a dict of peering info from a peering config section """
+            return {
+                # get name from `name[:type]/metanetwork
+                'vpc_name': endpoint.split('/')[0].split(':')[0].strip(),
+                # get type from `name[:type]/metanetwork`, defaulting to name if type is omitted
+                'vpc_type': endpoint.split('/')[0].split(':')[-1].strip(),
+                # get metanetwork from `name[:type]/metanetwork`
+                'metanetwork': endpoint.split('/')[1].strip()
+            }
+
+        def get_matching_vpcs(vpc_name, vpc_type, existing_vpcs):
+            """return a list of VPC names for VPCs that match the given vpc_name and vpc_type"""
+            return [vpc.tags['Name'] for vpc in existing_vpcs
+                    if vpc_name in ('*', vpc.tags['Name']) and vpc.tags['type'] == vpc_type]
+
+        source_info = get_endpoint_info(endpoints[0])
+        target_info = get_endpoint_info(endpoints[1])
+
+        if source_info['vpc_type'] == '*' or target_info['vpc_type'] == '*':
+            raise VPCConfigError('Wildcards are not allowed for VPC type in "%s". '
+                                 'Please specify a VPC type when using a wild card for the VPC name' % line)
```

Comment:
This statement might be a little misleading, since we don't seem to support wild card for VPC type at all.